src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/LanguageServerProjectLoader.cs

```diff
@@ -159,10 +159,12 @@ private async Task<bool> LoadOrReloadProjectAsync(ProjectToLoad projectToLoad, T
             if (preferredBuildHostKind != actualBuildHostKind)
                 preferredBuildHostKindThatWeDidNotGet = preferredBuildHostKind;
```

Comment:
We still need to move this block into the TryLoadProjectAsync() method, since the GetKindForProject is implicitly looking at the project file contents. In the case of the file-based programs, the selection is trivially "just use .NET Core".